action.yml

```diff
@@ -0,0 +1,141 @@
+name: 'build-ecs-nodejs'
+author: '1up-team'
+description: 'Action to build and release Nodejs & ECS repositories'
+
+inputs:
+  pkg-user:
+    description: 'User name used to integrate with the spring-media GitHub packages for internal libs'
+    required: true
+  pkg-token:
+    description: 'User token used to integrate with the spring-media GitHub packages for internal libs'
+    required: true
+
+  docker-image-tag:
+    description: 'Docker image build tag used when pushing to the 1up AWS ECR'
+    required: false
+    default: 'b${{ github.run_number }}-${{ github.sha }}'
+
+  github-token:
+    description: Github token, current token from secrets
+    required: false
+
+  staging:
+    description: 'Release to staging environment'
+    required: false
+    default: 'false'
+
+  up-slack-webhook-url:
+    description: 'Slack channel webhook URL used when reporting master build status'
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup - Validate Inputs
+      shell: bash
+      run: |
+        [[ "${{ inputs.pkg-user }}" ]] || { echo "input 'pkg-user' cannot be blank"; exit 1; }
+        [[ "${{ inputs.pkg-token }}" ]] || { echo "input 'pkg-token' cannot be blank"; exit 1; }
+
+    - name: Setup service name
+      run: |
+        SERVICE_NAME=${GITHUB_REPOSITORY##*/1up-}
+        echo "Service name: $SERVICE_NAME"
+        echo "SERVICE_NAME=$SERVICE_NAME" >> $GITHUB_ENV
+
+    - name: Build - Docker
+      run: |
+        DOCKER_IMAGE="${{ env.ECR_REPOSITORY }}/${{ env.SERVICE_NAME }}:${{ env.IMAGE_TAG }}"
+        docker build -t $DOCKER_IMAGE .
+        echo "DOCKER_IMAGE=$DOCKER_IMAGE" >> $GITHUB_ENV
+    - name: ︎Setup - AWS Credentials
+      uses: aws-actions/configure-aws-credentials@v1-node16
+      with:
+        aws-access-key-id: ${{ inputs.up-aws-access-key-id }}
+        aws-secret-access-key: ${{ inputs.up-aws-secret-access-key }}
+        aws-region: eu-west-1
+    - name: Release - push Docker image [on master]
+      run: docker push ${{ env.DOCKER_IMAGE }}
+
+    - uses: dorny/paths-filter@v2
+      id: infrastructure-changes
+      with:
+        filters: |
+          terraform:
+            - 'terraform/**'
+
+    - name: deploy infrastructure
+      if: steps.infrastructure-changes.outputs.terraform == 'true'
+      uses: spring-media/1up-github-actions@deploy-infrastructure
+      with:
+        service-name: ${{ env.SERVICE_NAME }}
+        pkg-user: ${{ inputs.pkg-user }}
```

Comment:
pkg-user is no longer required :)

Reply:
pkg-user being back?

Reply:
it was there since beginning, I'm still testing this pipeline

Reply:
not working with main one

Reply:
Just merged it :)